### DISCUSSION TITLE
Is there a way to delay the unmarshaling of a specific field?

### DISCUSSION THREAD

**Discussion by samhza** [pelletier/go-toml#624] [Questions] [answered]
For example, package `encoding/json` has its own RawMessage type which implements UnmarshalJSON but in reality just holds the raw JSON data so it can be unmarshaled later.

**Comment by pelletier** [comment, marked as answer]
That's a very interesting question! We don't have that today. I wonder how much sense it would make for TOML: a document has a different semantic meanings based on the order the elements appear in the document, and potentially may or may not be valid as a result. JSON does not have this problem.

Can you elaborate on a use-case for this?

**Reply by samhza** [reply]
I don't have a use-case for this, I was just wondering why go-toml doesn't have this functionality but now I got my answer. Thanks 👍
